cpp/src/arrow/sparse_tensor.h

```diff
@@ -220,14 +261,66 @@ class ARROW_EXPORT SparseCSRIndex : public internal::SparseIndexBase<SparseCSRIn
       return Status::OK();
     }
 
-    return Status::Invalid("shape length is inconsistent with the CSR index");
+    return Status::Invalid("shape length is inconsistent with the ", ToString());
   }
 
  protected:
   std::shared_ptr<Tensor> indptr_;
   std::shared_ptr<Tensor> indices_;
 };
 
+}  // namespace internal
+
+// ----------------------------------------------------------------------
+// SparseCSRIndex class
+
+/// \brief EXPERIMENTAL: The index data for a CSR sparse matrix
+///
+/// A CSR sparse index manages the location of its non-zero values by two
+/// vectors.
+///
+/// The first vector, called indptr, represents the range of the rows; the i-th
+/// row spans from indptr[i] to indptr[i+1] in the corresponding value vector.
+/// So the length of an indptr vector is the number of rows + 1.
+///
+/// The other vector, called indices, represents the column indices of the
+/// corresponding non-zero values.  So the length of an indices vector is same
+/// as the number of non-zero-values.
+class ARROW_EXPORT SparseCSRIndex
+    : public internal::SparseCSXIndex<SparseCSRIndex,
+                                      internal::SparseMatrixCompressedAxis::ROW> {
+ public:
+  constexpr static char const* TYPE_NAME = "SparseCSRIndex";
```

Comment:
Why not define `format_id` here instead of using `std::conditional` in the base class?

Reply:
I completely agree with you.
The reason is that I didn't make `SparseCSRIndex` and `SparseCSCIndex` as subclasses of `SparseCSXIndex` but just aliases of `SparseCSXIndex` in the version before pushing here.
I'll move `format_id` in the subclasses.